table/tables/tables.go

```diff
@@ -406,11 +389,19 @@ func (t *Table) rebuildIndices(rm kv.RetrieverMutator, h int64, touched map[int]
 }
 
 // AddRecord implements table.Table AddRecord interface.
-func (t *Table) AddRecord(ctx context.Context, r []interface{}, h int64) (recordID int64, err error) {
-	// Already have recordID
-	if h != 0 {
-		recordID = int64(h)
-	} else {
+func (t *Table) AddRecord(ctx context.Context, r []interface{}) (recordID int64, err error) {
+	var hasRecordID bool
+	for _, col := range t.Cols() {
+		if mysql.HasPriKeyFlag(col.Flag) && t.meta.PKIsHandle {
+			recordID, err = types.ToInt64(r[col.Offset])
```

Comment:
You have already converted r[col.Offset] value to int64 in initDefaultValues(). So why do it here again?